ast.rkt

```diff
@@ -130,16 +130,57 @@
      (table-identifier self))]
   #:methods gen:serializable
   [(define (serialize self #:size [_size #f])
+     (define (serialize-constraints constraint-list)
+       (define (serialize-constraint constraint)
+         (let* [(serialized-constraint (string->bytes/utf-8 (call-with-output-string
+                                                             (lambda (s-port)
+                                                               (write (syntax->datum constraint) s-port)))))
+                (constraint-size (call-with-output-bytes
+                                  (lambda (c-port)
+                                    (write-char (integer->char (bytes-length serialized-constraint)) c-port))))]
+           (bytes-append constraint-size serialized-constraint)))
+       (define constraints-count (length constraint-list))
+       (unless (<= constraints-count #xffff)
```

Comment:
```suggestion
       (unless (<= constraints-count #xff)
```
We neglected to change this after we reduced the number of bytes used to store this value.